src/Other/Helpers/Reflection.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Rubix\ML\Other\Helpers;
+
+use ReflectionClass;
+use ReflectionException;
+
+/**
+ * Reflection
+ *
+ * Utility functions relying on PHP's runtime reflection API
+ *
+ * @see https://www.php.net/manual/en/book.reflection.php
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Chris Simpson
+ */
+class Reflection
+{
+    /**
+     * Property.
+     *
+     * Access a class property even if it's visibility is not public.
+     * This is typically useful for debugging and creating informative logs.
+     *
+     * @param object $subject The object to introspect
+     * @param string $name The property to access
+     * @param mixed $fallback
+     * @return mixed
+     */
+    public static function property(object $subject, string $name, $fallback = null)
+    {
+        try {
+            $property = (new ReflectionClass($subject))->getProperty($name);
+            $property->setAccessible(true);
+
+            return $property->getValue($subject);
+        } catch (ReflectionException $e) {
+            return $fallback;
+        }
+    }
```

Comment:
I mentioned this method here: https://github.com/RubixML/RubixML/pull/109#discussion_r486620033. I had it in a local stash so thought i'd add it here.

I think it's the nicest way to keep the `Flysystem::__toString()` method clean, but retaining the useful functionality of exposing the specific `AdapterInterface` that is been used. 🤷‍♂️

Reply:
This is cool I like it